File: CS/CustomImageEditor/DragImageEdit/DragImagEditRepositoryItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Controls;
using DevExpress.Utils.Drawing;
using System.Drawing.Imaging;
using DevExpress.Utils;
using DevExpress.XtraPrinting;

namespace CustomImageEditor {
    [UserRepositoryItem("Register")]
    public class DragImagEditRepositoryItem : RepositoryItem {
   
        static DragImagEditRepositoryItem() {
            Register();
        }
        public static void Register() {
            EditorClassInfo myEditorCI = new EditorClassInfo(EditorName, typeof(DragImageEdit), typeof(DragImagEditRepositoryItem), typeof(DragImageEditViewInfo), new DragImagEditPainter(), true);
            EditorRegistrationInfo.Default.Editors.Add(myEditorCI);
        }
        internal const string EditorName = "myImageEdit";
        public override string EditorTypeName {
            get { return EditorName; }
        }
    }
}

File: VB/CustomImageEditor/DragImageEdit/DragImagEditRepositoryItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.Utils.Drawing
Imports System.Drawing.Imaging
Imports DevExpress.Utils
Imports DevExpress.XtraPrinting

Namespace CustomImageEditor
	<UserRepositoryItem("Register")> _
	Public Class DragImagEditRepositoryItem
		Inherits RepositoryItem

		Shared Sub New()
			Register()
		End Sub
		Public Shared Sub Register()
			Dim myEditorCI As New EditorClassInfo(EditorName, GetType(DragImageEdit), GetType(DragImagEditRepositoryItem), GetType(DragImageEditViewInfo), New DragImagEditPainter(), True)
			EditorRegistrationInfo.Default.Editors.Add(myEditorCI)
		End Sub
		Friend Const EditorName As String = "myImageEdit"
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
	End Class
End Namespace
